RHDEVDOCS-5792: Content creation for GitOps 1.12.0 RN

Dhruv-Soni11 opened the pull request:
This PR is for the GitOps 1.12 release that is scheduled for [?] day (not sure if this will be released either today or tomorrow). Do not merge until this date.

<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s):
<!--- Specify the version or versions of OpenShift your PR applies to. -->

GitOps 1.12

Issue:
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

https://issues.redhat.com/browse/RHDEVDOCS-5792

Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

https://72708--docspreview.netlify.app/openshift-gitops/latest/release_notes/gitops-release-notes

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

SME review has been completed by: @reginapizza 
QE review has been completed by: @varshab1210 
Internal peer review has been completed by: @shipsing  

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

Dhruv-Soni11 commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,95 @@
+[id="errata-updates-1-12.0_{context}"]
+== Errata updates
+
+TBD
```
@reginapizza - Do we need to document any Errata entries for this release? Keeping this section until you confirm. If there are no entries, I will delete this section.

reginapizza replied:
sorry, I'm not sure what you mean by errata entries?

reginapizza replied:
Hi @Dhruv-Soni11, this advisory has now been updated from a RHEA (Red Hat Enhancement Advisory) to a RHSA (Red Hat Security Advisory), so it will need that section now. Please coordinate with @Srivaralakshmi to understand how to get the info for that section as I'm a little unsure about that bit.

anandf commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,95 @@
+[id="new-features-1-12-0_{context}"]
+== New features
+
+* With this update, the Argo CD CLI is supported and accessible as a productized and easily-installable RPM. The Argo CD CLI tool enables users to manage Red Hat OpenShift GitOps from a terminal. Using this tool, users can manage ArgoCD resources, such as Application, ApplicationSets, AppProjects, User accounts, and GPG keys from a client.
```
users can manage ArgoCD resources (Application, ApplicationSets, AppProjects, User accounts, GPG keys etc) from a client terminal.

Dhruv-Soni11 replied:
Fixed.

varshab1210 commented on modules/go-compatibility-and-support-matrix.adoc:
@Dhruv-Soni11 Could you please fix this
![Screenshot from 2024-03-07 14-18-18](https://github.com/openshift/openshift-docs/assets/60093147/f997545b-8a33-49d5-8f26-4b957902622a)

Dhruv-Soni11 replied:
Fixed the table's formatting issues. Please confirm.

varshab1210 commented:
https://72708--docspreview.netlify.app/openshift-gitops/latest/release_notes/gitops-release-notes#GitOps-technology-preview_gitops-release-notes 

This table needs to be updated with ArgoCD CLI tool as well

varshab1210 commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,95 @@
+The Argo CD CLI tool is a Technology Preview feature only.
+====
+
+* With this update, two new fields `.spec. applicationSet.sourceNamespaces` and `.spec. applicationSet.scmProviders` are introduced in ArgoCD CRD to support `ApplicationSet` in any namespaces. Administrators use it to define certain namespaces that manage ApplicationSet resources.
```
```suggestion
* With this update, two new fields `.spec.applicationSet.sourceNamespaces` and `.spec.applicationSet.scmProviders` are introduced in ArgoCD CRD to support `ApplicationSet` in any namespaces. Administrators use it to define certain namespaces that manage ApplicationSet resources.
```

Dhruv-Soni11 replied:
Fixed.

varshab1210 commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,95 @@
+
+* With this update, Argo CD server has the required permissions to manage `ApplicationSet` resources.
+
+* With this update, you can configure log levels such as "debug", "info", "warn", "error", "panic" and "fatal". The default log level set for the output is "info". Users can select an appropriate log level by adding the environment variable `LOG_LEVEL` in Red Hat OpenShift GitOps subscription's field, `spec.config.env`.
```
```suggestion
* With this update, you can configure log levels such as "debug", "info", "warn", "error", "panic" and "fatal". The default log level set for the output is "info". Users can select an appropriate log level by adding the environment variable `LOG_LEVEL` in Red Hat OpenShift GitOps subscription's field, `.spec.config.env`.
```

Dhruv-Soni11 replied:
Fixed.

varshab1210 commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,95 @@
+      value: "error"
+----
+
+* With this feature, Argo CD uses the wildcard values in `sourceNamespaces` to specify multiple namespaces or patterns of namespaces. To use this feature, specify the namespaces where Argo CD can manage applications in the Argo CD YAML with `spec.sourceNamespaces`. 
```
```suggestion
* With this feature, Argo CD uses the wildcard values in `sourceNamespaces` to specify multiple namespaces or patterns of namespaces. To use this feature, specify the namespaces where Argo CD can manage applications in the Argo CD YAML with `.spec.sourceNamespaces`. 
```

Dhruv-Soni11 replied:
Fixed.

varshab1210 commented:
@Dhruv-Soni11 There is a known issue that we need to document 

https://issues.redhat.com/browse/GITOPS-4174

Dhruv-Soni11 commented:
> @Dhruv-Soni11 We usually add the Jira tickets after the description
> 
> Eg: https://72708--docspreview.netlify.app/openshift-gitops/latest/release_notes/gitops-release-notes#fixed-issues-1-11-1_gitops-release-notes

Fixed, Added JIRA ticket links after every Fixed Issue.

reginapizza commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,95 @@
+[id="gitops-release-notes-1-12-0_{context}"]
+= Release notes for {gitops-title} 1.12.0
+
+{gitops-title} 1.12.0 is now available on {OCP} 4.12, 4.13, and 4.14.
```
are we not doing 4.15? It's released now...

Dhruv-Soni11 replied:
It is released and I can add it too, pending further confirmation from you. I erred on the side of caution and did not include it. Please confirm if this release information can be included.

varshab1210 replied:
It seems like the announcement for 4.15 GA happens on March 19th so I believe e would need to make that change later

varshab1210 replied:
cc: @Srivaralakshmi

Dhruv-Soni11 replied:
Should I add that information now @varshab1210 or wait for the 19th although the release is going on March 14th. This is indeed a slight grey area. 

Tagging @iam-veeramalla and @reginapizza in this comment to get their views as well.

Dhruv-Soni11 replied:
> It seems like the announcement for 4.15 GA happens on March 19th so I believe e would need to make that change later

Hi @varshab1210 - As today is March 19th, I am including information about 4.15 in our RN. Please confirm if you are fine with this change. Thanks.

reginapizza commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,95 @@
+[id="new-features-1-12-0_{context}"]
+== New features
+
+* With this update, the Argo CD CLI is supported and accessible as a productized and easily-installable RPM. The Argo CD CLI tool enables users to manage Red Hat OpenShift GitOps from a terminal. Using this tool, users can manage ArgoCD resources, such as Application, ApplicationSets, AppProjects, User accounts, and GPG keys from a client.
```
Applications* (should be plural)

Dhruv-Soni11 replied:
Fixed.

reginapizza commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,95 @@
+
+* With this update, Argo CD server has the required permissions to manage `ApplicationSet` resources.
+
+* With this update, you can configure log levels such as "debug", "info", "warn", "error", "panic" and "fatal". The default log level set for the output is "info". Users can select an appropriate log level by adding the environment variable `LOG_LEVEL` in Red Hat OpenShift GitOps subscription's field, `spec.config.env`.
```
Not 100% sure of docs standards wrt release notes, but I think the log levels should be in the format: `debug`, `info`, `warn`, `error`, `panic` and `fatal`. (backticks instead of quotes) 

If you look here under ApplicationSet logLevel they are like that. https://docs.openshift.com/gitops/1.11/argocd_instance/argo-cd-cr-component-properties.html#argo-cd-properties_argo-cd-cr-component-properties

Dhruv-Soni11 replied:
Fixed.

reginapizza commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,95 @@
+[id="fixed-issues-1-12-0_{context}"]
+== Fixed issues
+
+* Before this update, the notifications controller could not use `built-in` functions in notification templates to obtain information about applications due to an incorrect repo-server address. This update fixes the issue by initializing the notification controller to use the correct repo-server address.
```
built-in is just a descriptor, I don't think it needs backticks

Dhruv-Soni11 replied:
Fixed.

reginapizza commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,95 @@
+
+* Before this update, when a user deletes apps and namespaces in Argo CD, in some cases, the Red Hat GitOps plugin displays an error message. This update fixes the issue by updating some components from the dynamic plugin SDK and provides better error handling. As a result, these error messages are not displayed after deleting an apps or namespaces.
+
+* Before this update, the password for an admin user was reset to the default password automatically when the Argo CD CLI or Web UI was updated. This update fixes the issue and the user can now update the admin password and the password is no longer set to default.
```
The original text was "password for admin user used to reset with default password when updated by ArgoCD CLI or UI"... it's a nuanced difference but the password being updated by the CLI/UI vs when the CLI/UI are updated are different

reginapizza commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,95 @@
+
+* Before this update, the notifications controller could not use `built-in` functions in notification templates to obtain information about applications due to an incorrect repo-server address. This update fixes the issue by initializing the notification controller to use the correct repo-server address.
+
+* In this update, users can go to the Argo CD Web UI and create job executions from a `cron` job defined in a deployed Argo CD application.
```
Sorry I know the example text we gave you didn't have this so I'm adding it now- generally under fixed issues in release notes we say what the issue was and then how it was fixed, otherwise it seems more like a feature. 

For this one, you can add something like "Before this update, when a user went to create a Job from a CronJob, they would get an error. Now, users can create job executions from a `cron` job defined in a deployed Argo CD application"

Dhruv-Soni11 replied:
Fixed.

reginapizza commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,95 @@
+
+* In this update, users can go to the Argo CD Web UI and create job executions from a `cron` job defined in a deployed Argo CD application.
+
+* In this update, the application environment card gets removed automatically after the Argo CD application is deleted. Refreshing or reloading of the Argo CD Web UI does not remove the application environment card.
```
same as above, need to add more context about the original issue... 

"Before this update, sometimes when applications are deleted, their cards would remain in a half-deleted state until the page was refreshed. Now the whole application environment card will be removed from the UI automatically after the Argo CD application is deleted."

Dhruv-Soni11 replied:
Fixed.

reginapizza commented:
One last comment for your PR overall- usually at the end of each release note item there is a link the jira ticket, formatted like this: `link:https://issues.redhat.com/browse/GITOPS-3922[GITOPS-3922]`. All of them are missing that, so please add them. Thanks!

Dhruv-Soni11 commented:
> One last comment for your PR overall- usually at the end of each release note item there is a link to the associated jira ticket, formatted like this: `link:https://issues.redhat.com/browse/GITOPS-3922[GITOPS-3922]`. All of them are missing that, so please add them. Thanks!

I thought only fixed issues needed the addition of associated JIRAs, which I have also added after the feedback provided by @varshab1210. I have now added the links for every associated RN entry. Thanks for pointing this out @reginapizza.

varshab1210 commented on modules/go-compatibility-and-support-matrix.adoc:
```diff
@@ -20,15 +20,17 @@ In {OCP} 4.13, the `stable` channel has been removed. Before upgrading to {OCP}
+|*Version* |*`kam`*    |*Helm*  |*Kustomize* |*Argo CD*|*Argo Rollouts*|*Dex*     |*RH SSO* |
 
-|1.11.0 |0.0.51 TP |3.13.2 GA |5.2.1 GA |2.9.2 GA |1.6.0 TP |NA |2.36.0 GA |7.6.0 GA |4.12-4.14
+|1.12.0 |0.0.51 TP |3.14.0 GA |5.2.1 GA |2.10.1 GA |1.6.0 TP |2.36.0 GA |7.6.0 GA |4.12-4.14
```
```suggestion
|1.12.0 |0.0.51 TP |3.14.0 GA |5.2.1 GA |2.10.3 GA |1.6.0 TP |2.36.0 GA |7.6.0 GA |4.12-4.14
```

Dhruv-Soni11 replied:
Fixed.

shipsing commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,97 @@
+
+* With this update, Argo CD server has the required permissions to manage `ApplicationSet` resources. link:https://issues.redhat.com/browse/GITOPS-3762[GITOPS-3762]
+
+* With this update, you can configure log levels such as `debug`, `info`, `warn`, `error`, `panic` and `fatal`. The default log level set for the output is `info`. Users can select an appropriate log level by adding the environment variable `LOG_LEVEL` in Red Hat OpenShift GitOps subscription's field, `.spec.config.env`. link:https://issues.redhat.com/browse/GITOPS-4016[GITOPS-4016]
```
```suggestion
* With this update, you can configure log levels, such as `debug`, `info`, `warn`, `error`, `panic` and `fatal`. The default log level set for the output is `info`. To change the log level, add the environment variable `LOG_LEVEL` in the `.spec.config.env` field of the `Subscription` CR. link:https://issues.redhat.com/browse/GITOPS-4016[GITOPS-4016]
```

Dhruv-Soni11 replied:
Fixed.

shipsing commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,97 @@
+.Example
+[source,yaml]
+----
+   yaml
```
do we need this line?? We usually start from the `apiVersion` field.

shipsing commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,97 @@
+.Example
+[source,yaml]
+----
+yaml
```
same comment
do we need this line?? We usually start from the `apiVersion` field.

Dhruv-Soni11 replied:
Fixed, removing YAML as per your suggestion.

shipsing commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,97 @@
+
+* Before this update, the notifications controller could not use built-in functions in notification templates to obtain information about applications due to an incorrect repo-server address. This update fixes the issue by initializing the notification controller to use the correct repo-server address. link:https://issues.redhat.com/browse/GITOPS-2867[GITOPS-2867]
+
+* Before this update, when a user created a `Job` from a `CronJob`, an error would appear. Now, users can create `Job` executions from a `CronJob` defined in a deployed Argo CD application. link:https://issues.redhat.com/browse/GITOPS-3617[GITOPS-3617]
```
What is the entity type for `CronJob`? is it an object or resource or anything else?

Dhruv-Soni11 replied:
Fixed, changing the term from `CronJob` to `CronJob` resource.

shipsing commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,97 @@
+
+* Before this update, in some instance when applications are deleted, their application environment card would remain in a half-deleted state until the page was refreshed. Now, the application environment card will be removed from the UI automatically after the Argo CD application is deleted. link:https://issues.redhat.com/browse/GITOPS-2677[GITOPS-2677]
+
+* Before this update, when a user deletes apps and namespaces in Argo CD, in some cases, the Red Hat GitOps plugin displays an error message. This update fixes the issue by updating some components from the dynamic plugin SDK and provides better error handling. As a result, these error messages are not displayed after deleting an apps or namespaces. link:https://issues.redhat.com/browse/GITOPS-27 ...
```
```suggestion
* Before this update, when a user deleted apps and namespaces in Argo CD, in some cases, the {gitops-title} plugin would display an error message. This update fixes the issue by updating some components from the dynamic plugin SDK and provides better error handling. As a result, the error message is not displayed after deleting apps or namespaces. link:https://issues.redhat.com/browse/GITOPS-2746[GITOPS-2746]
```

Dhruv-Soni11 replied:
Fixed.

shipsing commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,97 @@
+      value: "error"
+----
+
+* With this feature, Argo CD uses the wildcard values in `sourceNamespaces` to specify multiple namespaces or patterns of namespaces. To use this feature, specify the namespaces where Argo CD can manage applications in the Argo CD YAML with `.spec.sourceNamespaces`. link:https://issues.redhat.com/browse/GITOPS-3935[GITOPS-3935]
```
```suggestion
* With this update, Argo CD accepts the wildcard values in the `sourceNamespaces` field so that you can specify multiple namespaces or patterns of namespaces. To use this feature, specify the namespaces where Argo CD can manage applications in the `.spec.sourceNamespaces` field of the `ArgoCD` CR. link:https://issues.redhat.com/browse/GITOPS-3935[GITOPS-3935]
```

Dhruv-Soni11 replied:
Fixed.

shipsing commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,97 @@
+
+In this example, permissions are granted to namespaces that match the pattern `app-team-*`, such as `app-team-1`, `app-team-2`, and namespace-2, which does not use wildcard values.
+
+To grant permissions for all the namespaces on the Argo CD cluster using the `*` wildcard pattern, the following Argo CD CR configuration must be used:
```
```suggestion
To grant permissions for all the namespaces on the Argo CD cluster using the `*` wildcard pattern, configure the `ArgoCD` CR in the following manner:
```

Dhruv-Soni11 replied:
Fixed.

shipsing commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,97 @@
+.Example
+[source,yaml]
+----
+yaml
```
do we need this line?? We usually start from the `apiVersion` field.

Dhruv-Soni11 replied:
Fixed.

Dhruv-Soni11 commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,97 @@
+
+* With this update, users can configuration notifications in a dedicated `NotificationsConfiguration` custom resource (CR). This process replaces the workflow in earlier releases of {gitops-title}. To configure this feature in earlier releases, you had to edit the `argocd-notifications-cm` ConfigMap manually. Users can now set Triggers, Templates, Services, and Subscriptions as fields in the new  ...
+
+* With this update, two new fields `.spec.applicationSet.sourceNamespaces` and `.spec.applicationSet.scmProviders` are introduced in ArgoCD CRD to support `ApplicationSet` in any namespaces. Administrators use it to define certain namespaces that manage `ApplicationSet` resources. link:https://issues.redhat.com/browse/GITOPS-3754[GITOPS-3754]
```
Fixed.

Dhruv-Soni11 commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,97 @@
+
+[WARNING]
+==== 	
+The Argo CD CLI tool is a Technology Preview feature only.
```
I think this is a good point that we can add in a different section called as [Technology Preview Features](https://docs.openshift.com/gitops/1.10/release_notes/gitops-release-notes.html#GitOps-technology-preview_gitops-release-notes) in our current documentation. However, I would only like to implement it if @Srivaralakshmi feels the same as we have not added anything like this for other similar notes in the GitOps RN.

WDYT about this @Srivaralakshmi?

Srivaralakshmi replied:
@shipsing and @Dhruv-Soni11 The support scope is already mentioned here: https://docs.openshift.com/gitops/1.11/release_notes/gitops-release-notes.html#GitOps-compatibility-support-matrix_gitops-release-notes. 

See the 1st line: Some features in this release are currently in [Technology Preview](https://access.redhat.com/support/offerings/techpreview).

As short understanding about TP feature is here: https://docs.openshift.com/gitops/1.11/release_notes/gitops-release-notes.html#GitOps-technology-preview_gitops-release-notes

shipsing replied:
yes, @Dhruv-Soni11

Dhruv-Soni11 replied:
Thanks for the confirmation @shipsing. Marking this comment as resolved.

reginapizza commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,94 @@
+The Argo CD CLI tool is a Technology Preview feature.
+====
+
+* With this update, users can configure notifications in a dedicated `NotificationsConfiguration` custom resource (CR). This configuration replaces the workflow in earlier releases of {gitops-title}. To configure notifications in earlier releases, you had to edit the `argocd-notifications-cm` ConfigMap manually. Users can now set `Triggers`, `Templates`, `Services`, and `Subscriptions` as fields  ...
```
There is a note in the release note text for this [here](https://docs.google.com/document/d/1_3Hjsc3rAGvmjNo2GY_oYGqEzeyNusEOes9DNMoar9Y/edit). I think we should add something here like: 

```
Note - This update removes any configuration added to `argocd-notifications-cm` ConfigMap. We recommend users to take a backup of their configuration and update that in `default-notifications-configuration` custom resource of Kind `NotificationsConfiguration` after upgrading to the new version. Also with this update, any modifications to `argocd-notifications-cm`ConfigMap are not allowed.

```

Dhruv-Soni11 replied:
Thanks for providing this information @reginapizza. I have added this note after the content here.

Srivaralakshmi commented on modules/go-compatibility-and-support-matrix.adoc:
```diff
@@ -20,15 +20,17 @@ In {OCP} 4.13, the `stable` channel has been removed. Before upgrading to {OCP}
-|1.9.0    |0.0.49 TP |3.11.2 GA|5.0.1 GA   |2.7.2 GA |1.5.0 TP |NA     |2.35.1 GA |7.5.1 GA |4.12-4.14
+|1.10.0 |0.0.50 TP |3.12.1 GA |5.1.0 GA |2.8.3 GA |1.5.0 TP |2.35.1 GA |7.5.1 GA |4.12-4.14
+
+|1.9.0  |0.0.49 TP |3.11.2 GA |5.0.1 GA |2.7.2 GA |1.5.0 TP |2.35.1 GA |7.5.1 GA |4.12-4.14
```
@harrietgrace @iam-veeramalla @reginapizza @varshab1210 Should we remove 1.9.0 from the table as we support only n, n-1, and n-2 versions?

varshab1210 replied:
Yes, we would need to remove it
cc: @Dhruv-Soni11

Dhruv-Soni11 replied:
Fixed.

Srivaralakshmi commented:
@Dhruv-Soni11 Please update the [Technology Preview tracker table](https://72708--docspreview.netlify.app/openshift-gitops/latest/release_notes/gitops-release-notes#GitOps-technology-preview_gitops-release-notes) for [#3754](https://issues.redhat.com/browse/GITOPS-3754)- Support applicationsets in any namespace and other TP features if any for this release.

Dhruv-Soni11 commented:
peer-review-needed

shipsing commented:
/label peer-review-needed

xenolinux commented:
/label peer-review-in-progress

xenolinux commented:
@Dhruv-Soni11 Is this PR ready for peer review? By looking at "TBD" it seems like PR is expecting some changes.

xenolinux commented:
/remove-label peer-review-needed
/remove-label peer-review-in-progress

Dhruv-Soni11 commented:
> Argo CD application sets in non-control plane namespaces

Fixed it as per our discussion on slack @Srivaralakshmi. Thanks for the inputs.

dfitzmau commented:
/remove-label peer-review-needed

/label peer-review-in-progress

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,115 @@
+[id="new-features-1-12-0_{context}"]
+== New features
+
+* With this update, the Argo CD CLI is supported and accessible as a productized and easily-installable RPM. The Argo CD CLI tool enables users to manage {gitops-title} from a terminal. Using this tool, users can manage ArgoCD resources, such as `Applications`, `ApplicationSets`, `AppProjects`, `User accounts`, and `GPG keys` from a client terminal. link:https://issues.redhat.com/browse/GITOPS-33 ...
```
According to the IBM SG, use words like "easy" are best reserved when making comparisons.

dfitzmau replied:
>The Argo CD CLI tool enables users

"Enables" is a form of anthrophomorphism:

![Screenshot from 2024-03-12 16-22-36](https://github.com/openshift/openshift-docs/assets/57954076/b5663002-5840-4e95-829f-a83eff71dce7)

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,115 @@
+
+[NOTE]
+====
+The `argocd` executable binary file is included in the archives and the RPMs.
```
```suggestion
The `argocd` executable binary file is included in the archive and RPM formats.
```

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,115 @@
+The Argo CD CLI tool is a Technology Preview feature.
+====
+
+* With this update, users can configure notifications in a dedicated `NotificationsConfiguration` custom resource (CR). This configuration replaces the workflow in earlier releases of {gitops-title}. To configure notifications in earlier releases, you had to edit the `argocd-notifications-cm` ConfigMap manually. Users can now set `Triggers`, `Templates`, `Services`, and `Subscriptions` as fields  ...
```
users/you ? Why remove the "you" perspective?

dfitzmau replied:
Any way of avoiding stacked admonitions? Maybe use the following example to build "Technology Preview" into the title?

![Screenshot from 2024-03-12 16-31-13](https://github.com/openshift/openshift-docs/assets/57954076/b3f8badb-cc0a-43ec-9333-d87f04545ca7)

![Screenshot from 2024-03-12 16-32-06](https://github.com/openshift/openshift-docs/assets/57954076/826a5bd4-35d7-4572-9c5e-70c8ab6673ab)

dfitzmau replied:
Indent a note by using the plus "+" character.

Dhruv-Soni11 replied:
> users/you ? Why remove the "you" perspective?

Fixed. Trying to make it as a standard "you" perspective throughout the RN content.

Dhruv-Soni11 replied:
> Indent a note by using the plus "+" character.

Fixed.

Dhruv-Soni11 replied:
> Any way of avoiding stacked admonitions? Maybe use the following example to build "Technology Preview" into the title?
> 
> ![Screenshot from 2024-03-12 16-31-13](https://private-user-images.githubusercontent.com/57954076/312161484-b3f8badb-cc0a-43ec-9333-d87f04545ca7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzMTEwMTUsIm5iZiI6MTcxMDMxMDcxNSwicGF0aCI6Ii81Nzk1NDA3Ni8zMTIxNjE0ODQtYjNmOGJhZGItY2MwYS00M2VjLTkzMzMtZDg3ZjA0NTQ1Y2E3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEzVDA2MTgzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcxNTIwODQwNmQ0ZjJhNjkzZmU5ZjE5OGM5MjkxZjliNmUzZDY5M2U5MDY2MjdkMjhhNmMyZTM0NTA3NDBkYzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.I45z-EgBVVo0tETdhDQCJP-_blsnACICUonbe8nqexM)
> 
> ![Screenshot from 2024-03-12 16-32-06](https://private-user-images.githubusercontent.com/57954076/312161487-826a5bd4-35d7-4572-9c5e-70c8ab6673ab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzMTEwMTUsIm5iZiI6MTcxMDMxMDcxNSwicGF0aCI6Ii81Nzk1NDA3Ni8zMTIxNjE0ODctODI2YTViZDQtMzVkNy00NTcyLTljNWUtNzBjOGFiNjY3M2FiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEzVDA2MTgzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdkMjVkOGVjZjgyODJiMDJmZTllYzlkMWQ4NjZmYzcwYmFlOTc4NTI5ZWE0YWZiNzYyNGZjMmExMTAxYTAzMWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wHjj2lwYNP-BjfyTmRoCDYmdL4XTqXMu2S2V7Hzo77M)

@dfitzmau - I am unsure about how to implement your suggestion?. Could you please elaborate?

dfitzmau replied:
With this update, the Argo CD CLI is supported and accessible as a productized and is an easier method for installation because it uses an RPM binary versus XM. 
+
TECH PREVIEW warning
+
By using the Argo CD CLI tool, you can complete the following tasks:
+
Manage Red Hat OpenShift GitOps from a terminal.
Manage ArgoCD resources, such as Applications, ApplicationSets, AppProjects, User accounts, and GPG keys from a client terminal. [GITOPS-3389](https://issues.redhat.com/browse/GITOPS-3389)
+
NOTE

dfitzmau replied:
How about this to avoid two notes on top of each other (stacked notes).

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,115 @@
+The Argo CD CLI tool is a Technology Preview feature.
+====
+
+* With this update, users can configure notifications in a dedicated `NotificationsConfiguration` custom resource (CR). This configuration replaces the workflow in earlier releases of {gitops-title}. To configure notifications in earlier releases, you had to edit the `argocd-notifications-cm` ConfigMap manually. Users can now set `Triggers`, `Templates`, `Services`, and `Subscriptions` as fields  ...
```
This release note takes a reverse approach to the before:now format. 

See https://redhat-documentation.github.io/supplementary-style-guide/#_bug_fix

Dhruv-Soni11 replied:
This is an important feature in the release and I want to make sure that the users understand it, so I have written it in this format. Could you let me know how to improve this with the approach you are suggesting?

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,115 @@
+[id="fixed-issues-1-12-0_{context}"]
+== Fixed issues
+
+* Before this update, the notifications controller could not use built-in functions in notification templates to obtain information about applications due to an incorrect repo-server address. This update fixes the issue by initializing the notification controller to use the correct repo-server address. link:https://issues.redhat.com/browse/GITOPS-2867[GITOPS-2867]
```
repo-server correct? Long hand form better (repository server)?

Dhruv-Soni11 replied:
Yes, I agree with you. The long hand form is indeed better. Fixing it as per your suggestion.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,115 @@
+
+* Before this update, the notifications controller could not use built-in functions in notification templates to obtain information about applications due to an incorrect repo-server address. This update fixes the issue by initializing the notification controller to use the correct repo-server address. link:https://issues.redhat.com/browse/GITOPS-2867[GITOPS-2867]
+
+* Before this update, when a user created a `Job` from a `CronJob` resource, an error would appear. Now, users can create `Job` executions from a `CronJob` resource defined in a deployed Argo CD application. link:https://issues.redhat.com/browse/GITOPS-3617[GITOPS-3617]
```
![Screenshot from 2024-03-12 17-00-47](https://github.com/openshift/openshift-docs/assets/57954076/17dce206-fc61-498a-9da6-60fc4d74ab0c)

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented:
Hello @Dhruv-Soni11 . Nice work on the release notes. I added comments inline. Feel free to contact me if you have any questions. My review was based on comparison with the core-COP format, but maybe GitOps has a different approach.

/remove-label peer-review-in-progress

dfitzmau commented:
/label peer-review-done

Dhruv-Soni11 commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,115 @@
+[id="new-features-1-12-0_{context}"]
+== New features
+
+* With this update, the Argo CD CLI is supported and accessible as a productized and easily-installable RPM. The Argo CD CLI tool enables users to manage {gitops-title} from a terminal. Using this tool, users can manage ArgoCD resources, such as `Applications`, `ApplicationSets`, `AppProjects`, `User accounts`, and `GPG keys` from a client terminal. link:https://issues.redhat.com/browse/GITOPS-33 ...
```
Fixed.

Dhruv-Soni11 commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,115 @@
+
+[WARNING]
+==== 	
+The Argo CD CLI tool is a Technology Preview feature.
```
Fixed.

Dhruv-Soni11 commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,115 @@
+
+[NOTE]
+====
+This update removes any configuration added to `argocd-notifications-cm` ConfigMap. Users must take a backup of their configuration and update that in the `default-notifications-configuration` custom resource of kind `NotificationsConfiguration` after upgrading to the new version. Also, with this update, any modifications to `argocd-notifications-cm` ConfigMap are not allowed.
```
Fixed, added a plus symbol before the note to ensure that the indentation is accurate.

Dhruv-Soni11 commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,115 @@
+This update removes any configuration added to `argocd-notifications-cm` ConfigMap. Users must take a backup of their configuration and update that in the `default-notifications-configuration` custom resource of kind `NotificationsConfiguration` after upgrading to the new version. Also, with this update, any modifications to `argocd-notifications-cm` ConfigMap are not allowed.
+====
+
+* With this update, two new fields `.spec.applicationSet.sourceNamespaces` and `.spec.applicationSet.scmProviders` are introduced in the `ArgoCD` CRD to support `ApplicationSet` in any namespaces. Administrators can use these fields to define certain namespaces that manage `ApplicationSet` resources. link:https://issues.redhat.com/browse/GITOPS-3754[GITOPS-3754]
```
I've checked a few previous versions of the GitOps release note and I have not found any instances of bold being applied to the JIRA name. Could I continue with the same format for my current release notes?

dfitzmau replied:
Hi @Dhruv-Soni11 . Yes, you can skip this as GitOps has its own way, by the looks of things, for formatting link metadata.

Dhruv-Soni11 commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,115 @@
+* With this update, Argo CD server has the required permissions to manage `ApplicationSet` resources. link:https://issues.redhat.com/browse/GITOPS-3762[GITOPS-3762]
+
+* With this update, you can configure log levels, such as `debug`, `info`, `warn`, `error`, `panic` and `fatal`. The default log level set for the output is `info`. To change the log level, add the environment variable `LOG_LEVEL` in the `.spec.config.env` field of the `Subscription` CR. link:https://issues.redhat.com/browse/GITOPS-4016[GITOPS-4016]
+
```
Fixed.

Dhruv-Soni11 commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,115 @@
+   - namespace-2
+----
+
+In this example, permissions are granted to namespaces that match the pattern `app-team-*`, such as `app-team-1`, `app-team-2`, and namespace-2, which does not use wildcard values.
```
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+[id="new-features-1-12-0_{context}"]
+== New features
+
+* With this update, the Argo CD CLI is supported and accessible as a productized and is an easier method for installation because it uses an RPM binary versus XM. By using the Argo CD CLI tool, you can complete the following tasks:
```
productized is missing something. Should it state a "productized version of something"?

"versus XM" is intentional until an SME confirm ation?

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+The Argo CD CLI tool is a Technology Preview feature.
+====
+
+* With this update, `NotificationsConfiguration` custom resource (CR) is now supported. Before this update, you could update the `argocd-notifications-cm` ConfigMap to manage their templates, triggers, services and subscriptions. After this update, you cannot modify the `argocd-notifications-cm` ConfigMap. A new Custom Resource definition `NotificationsConfiguration` is introduced to manage the ` ...
```
```suggestion
* With this update, `NotificationsConfiguration` custom resource (CR) is now supported. Before this update, you had to update the `argocd-notifications-cm` ConfigMap to manage templates, triggers, services and subscriptions. With this release, you cannot modify the `argocd-notifications-cm` ConfigMap. A new Custom Resource definition `NotificationsConfiguration` is introduced to manage the `argocd-notifications-cm` ConfigMap. link:https://issues.redhat.com/browse/GITOPS-4130[GITOPS-4130]
```

dfitzmau replied:
A customer can update `NotificationsConfiguration`?

Dhruv-Soni11 replied:
Yes, as per the information provided in the release notes document. Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
++
+[NOTE]
+====
+This update removes any configuration added to `argocd-notifications-cm` ConfigMap. Users must take a backup of their configuration and update that in the `default-notifications-configuration` custom resource of kind `NotificationsConfiguration` after upgrading to the new version. Also, with this update, any modifications to `argocd-notifications-cm` ConfigMap are not allowed.
```
```suggestion
This update removes any configuration added to `argocd-notifications-cm` ConfigMap. You must take a backup of your configuration and update the configuration/backup in the `default-notifications-configuration` custom resource of `kind:NotificationsConfiguration` after upgrading to the new version. Also, with this update, any modifications to `argocd-notifications-cm` ConfigMap are not allowed.
```

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+      value: "error"
+----
+
+* With this update, Argo CD accepts the wildcard values in the `sourceNamespaces` field so that you can specify multiple namespaces or patterns of namespaces. To use this feature, specify the namespaces where Argo CD can manage applications in the `.spec.sourceNamespaces` field of the `ArgoCD` CR. link:https://issues.redhat.com/browse/GITOPS-3935[GITOPS-3935]
```
```suggestion
* With this update, Argo CD accepts the wildcard values in the `sourceNamespaces` field so that you can specify multiple namespaces or patterns for namespaces. To use this feature, specify the namespaces where Argo CD can manage applications in the `.spec.sourceNamespaces` field of the `ArgoCD` CR. link:https://issues.redhat.com/browse/GITOPS-3935[GITOPS-3935]
```

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+    - app-team-*
+   - namespace-2
+----
+
```
Add a plus (+) before each paragraph to fix the indentation and the floating plus.

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+[id="fixed-issues-1-12-0_{context}"]
+== Fixed issues
+
+* Before this update, the notifications controller could not use built-in functions in notification templates to obtain information about applications due to an incorrectrepository server address. This update fixes the issue by initializing the notification controller to use the correct repository server address. link:https://issues.redhat.com/browse/GITOPS-2867[GITOPS-2867]
```
```suggestion
* Before this update, the notifications controller could not use built-in functions in notification templates to obtain information about applications because of an incorrect repository server address. This update fixes the issue by initializing the notification controller to use the correct repository server address. link:https://issues.redhat.com/browse/GITOPS-2867[GITOPS-2867]
```

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+
+* Before this update, the notifications controller could not use built-in functions in notification templates to obtain information about applications due to an incorrectrepository server address. This update fixes the issue by initializing the notification controller to use the correct repository server address. link:https://issues.redhat.com/browse/GITOPS-2867[GITOPS-2867]
+
+* Before this update, when a user created a `Job` from a `CronJob` resource, an error is displayed. Now, users can create `Job` executions from a `CronJob` resource defined in a deployed Argo CD application. link:https://issues.redhat.com/browse/GITOPS-3617[GITOPS-3617]
```
Some reviewers get hung up with the word "Now", so maybe follow the RH SSG example: 

![Screenshot from 2024-03-13 09-19-28](https://github.com/openshift/openshift-docs/assets/57954076/5cdac652-8f9b-459f-be67-e98e3422e0a8)

Dhruv-Soni11 replied:
Thanks for pointing this out Darragh. Added feedback.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+
+* Before this update, when a user created a `Job` from a `CronJob` resource, an error is displayed. Now, users can create `Job` executions from a `CronJob` resource defined in a deployed Argo CD application. link:https://issues.redhat.com/browse/GITOPS-3617[GITOPS-3617]
+
+* Before this update, in some instance when applications are deleted, their application environment card would remain in a half-deleted state until the page was refreshed. Now, the application environment card will be removed from the UI automatically after the Argo CD application is deleted. link:https://issues.redhat.com/browse/GITOPS-2677[GITOPS-2677]
```
```suggestion
* Before this update, in some instance when applications are deleted, their application environment card would remain in a half-deleted state until the page was refreshed. Now, the application environment card is removed from the UI automatically after the Argo CD application is deleted. link:https://issues.redhat.com/browse/GITOPS-2677[GITOPS-2677]
```

![Screenshot from 2024-03-13 09-20-53](https://github.com/openshift/openshift-docs/assets/57954076/6e25dfc0-c619-4d7c-a4fb-485a026c1b19)

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+
+* Before this update, when a user deleted apps and namespaces in Argo CD, in some cases, the {gitops-title} plugin would display an error message. This update fixes the issue by updating some components from the dynamic plugin SDK and provides better error handling. As a result, the error message is not displayed after deleting apps or namespaces. link:https://issues.redhat.com/browse/GITOPS-2746 ...
+
+* Before this update, the password for admin user used to reset to the default password when it was updated by the Argo CD CLI or the Argo CD UI. This update fixes the issue and the user can now update the admin password and the password is no longer set to default. link:https://issues.redhat.com/browse/GITOPS-3581[GITOPS-3581]
```
```suggestion
* Before this update, the password for the `admin` role was used to reset to the default password when it was updated by the Argo CD CLI or the Argo CD UI. This update fixes the issue and the user can now update the admin password and the password is no longer set to default. link:https://issues.redhat.com/browse/GITOPS-3581[GITOPS-3581]
```

dfitzmau replied:
![Screenshot from 2024-03-13 09-23-15](https://github.com/openshift/openshift-docs/assets/57954076/865bf03f-7f58-4eb1-a2ed-1434eb2128e9)

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+[id="known-issues-1-12-0_{context}"]
+== Known Issues
+
+* There is currently a known issue that changes the functionality of routing in the OpenShift Console Dynamic plugin, `dynamic-console-sdk`, that is used by the {gitops-title} Dynamic Plugin. This issue causes the horizontal navigation bar, which is used to switch between the Application Overview page and the Deployment History page for an application, on the Application Overview page to not func ...
```
```suggestion
* There is currently a known issue that changes the functionality of routing in the OpenShift Console Dynamic plugin, `dynamic-console-sdk`, that is used by the {gitops-title} Dynamic Plugin. This issue causes the horizontal navigation bar, which is used to switch between the *Application Overview* page and the *Deployment History* page for an application, on the *Application Overview* page to not function correctly.
```

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+
+* There is currently a known issue that changes the functionality of routing in the OpenShift Console Dynamic plugin, `dynamic-console-sdk`, that is used by the {gitops-title} Dynamic Plugin. This issue causes the horizontal navigation bar, which is used to switch between the Application Overview page and the Deployment History page for an application, on the Application Overview page to not func ...
++
+Workaround: To view the Deployment History of an application, use the *Deployment History* link on the application instead of the link on the horizontal navigation bar on the Application Overview page. This issue has been observed on all supported cluster versions that use the {gitops-title} Dynamic Plugin.
```
```suggestion
Workaround: To view the **Deployment History** page of an application, use the *Deployment History* link on the application instead of the link on the horizontal navigation bar on the *Application Overview* page. This issue has been observed on all supported cluster versions that use the {gitops-title} Dynamic Plugin.
```

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
++
+Workaround: To view the Deployment History of an application, use the *Deployment History* link on the application instead of the link on the horizontal navigation bar on the Application Overview page. This issue has been observed on all supported cluster versions that use the {gitops-title} Dynamic Plugin.
+
+* There is currently a known issue on the {OCP} 4.15 cluster for the console {gitops-title} Dynamic Plugin where users may experience some delay when fetching application data.
```
Just checking that GitOps does not have an attribute for 4.15?

dfitzmau replied:
may/might

![Screenshot from 2024-03-13 09-26-08](https://github.com/openshift/openshift-docs/assets/57954076/ffa4de84-845e-49c6-b699-6ecff07a6d51)

Dhruv-Soni11 replied:
Unsure about the 4.15 attribute, will confirm it with my reviewer. Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+
+* There is currently a known issue on the {OCP} 4.15 cluster for the console {gitops-title} Dynamic Plugin where users may experience some delay when fetching application data.
++
+Workaround: There is currently no workaround for this issue. It is recommended to wait for fetching the application data.
```
"Recommendations" can be tricky and the IBM SG states to avoid them. How about:

"No workaround currently exists for this issue, so you must wait for the fetching of the application data to complete".

Dhruv-Soni11 replied:
Nice suggestion, fixing it as per your feedback.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+* There is currently a known issue that causes the {gitops-title} Dynamic Plugin to be disabled on the {OCP} 4.15 cluster.
++
+Workaround: Perform the following steps:
+
```
Indent with pluses.

+
Workaround...
+
NOTE
+
Procedure step 1

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+Workaround: Perform the following steps:
+
+[NOTE]
+====
```
Maybe the note could be removed. Instead add the sentence to the procedure lead-in sentence.

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+====
+
+. In the *Administrator* perspective of the web console, navigate to *Home* -> *Overview*.
+. On the Overview tab, click the `Dynamic plugins` link in the Status section. 
```
```suggestion
. On the *Overview* tab, click the *Dynamic plugins* link in the *Status* section. 
```

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,143 @@
+
+. In the *Administrator* perspective of the web console, navigate to *Home* -> *Overview*.
+. On the Overview tab, click the `Dynamic plugins` link in the Status section. 
+. To enable the {gitops-title} Dynamic Plugin, click `gitops-plugin` and then click `Enabled`. 
```
```suggestion
. To enable the {gitops-title} Dynamic Plugin, click *gitops-plugin* and then click *Enabled*. 
```

Dhruv-Soni11 replied:
Fixed.

Dhruv-Soni11 commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,115 @@
+[id="gitops-1-12-0-security-update-advisory_{context}"]
+=== RHSA-xxxx:nnnn - {gitops-title} 1.12.0 security update advisory
+
+Issued: 2024-03-14
```
Hey Darragh, the details are not available to me yet. I will populate it once I get more information from my fellow peer writer.

dfitzmau commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,138 @@
+. In the *Administrator* perspective of the web console, navigate to *Home* -> *Overview*.
+. On the *Overview* tab, click the *Dynamic plugins* link in the *Status* section. 
+. To enable the {gitops-title} Dynamic Plugin, click *gitops-plugin* and then click *Enabled*. 
+
```
Add a plus before to indent the sentence:

Some time, a notification with the message “Web console update is available” is displ

Dhruv-Soni11 replied:
Fixed.

Srivaralakshmi commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,138 @@
+* With this update, `NotificationsConfiguration` custom resource (CR) is now supported. Before this update, you had to update the `argocd-notifications-cm` ConfigMap to manage templates, triggers, services and subscriptions. With this release, you cannot modify the `argocd-notifications-cm` ConfigMap. A new Custom Resource definition `NotificationsConfiguration` is introduced to manage the `argoc ...
+[NOTE]
+====
+This update removes any configuration added to `argocd-notifications-cm` ConfigMap. You must take a backup of your configuration and update the configuration/backup in the `default-notifications-configuration` custom resource of `kind:NotificationsConfiguration` after upgrading to the new version. Also, with this update, any modifications to `argocd-notifications-cm` ConfigMap are not allowed.
```
+1 to @dfitzmau

Dhruv-Soni11 replied:
Fixed.

dfitzmau commented:
Thanks, @Dhruv-Soni11 , for fixing the formatting bits! LGTM

reginapizza commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,142 @@
++
+Workaround: To view the **Deployment History** page of an application, use the *Deployment History* link on the application instead of the link on the horizontal navigation bar on the *Application Overview* page. This issue has been observed on all supported cluster versions that use the {gitops-title} Dynamic Plugin.
+
+* There is currently a known issue on the {OCP} 4.15 cluster for the console {gitops-title} Dynamic Plugin where users can experience some delay when fetching application data. link:https://issues.redhat.com/browse/GITOPS-4234[GITOPS-4234]
```
Yeah I agree, it does apply to 4.14 as well. @Dhruv-Soni11

Dhruv-Soni11 replied:
Fixed.

Srivaralakshmi commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,142 @@
+== Errata updates
+
+[id="gitops-1-12-0-security-update-advisory_{context}"]
+=== RHSA-xxxx:nnnn - {gitops-title} 1.12.0 security update advisory
```
@Dhruv-Soni11 Errata info for now to get updated in the RN:

- RHSA-2024:127897

TY!

Dhruv-Soni11 replied:
Fixed.

Dhruv-Soni11 commented:
This PR is for the GitOps 1.12 release that is scheduled for [?] day (not sure if this will be released either today or tomorrow). Do not merge until this date.

varshab1210 commented:
LGTM, thanks

Dhruv-Soni11 commented:
/label merge-review-needed

ShaunaDiaz commented:
/remove-label merge-review-needed

Srivaralakshmi commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,142 @@
+[id="errata-updates-1-12.0_{context}"]
+== Errata updates
+
+[id="RHSA-2024:127897-gitops-1-12-0-security-update-advisory_{context}"]
```
```suggestion
[id="RHSA-2024:1441-gitops-1-12-0-security-update-advisory_{context}"]
```

Dhruv-Soni11 replied:
Fixed.

Srivaralakshmi commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,142 @@
+== Errata updates
+
+[id="RHSA-2024:127897-gitops-1-12-0-security-update-advisory_{context}"]
+=== RHSA-xxxx:nnnn - {gitops-title} 1.12.0 security update advisory
```
```suggestion
=== RHSA-2024:1441 - {gitops-title} 1.12.0 security update advisory
```

Dhruv-Soni11 replied:
Fixed.

Srivaralakshmi commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,142 @@
+[id="RHSA-2024:127897-gitops-1-12-0-security-update-advisory_{context}"]
+=== RHSA-xxxx:nnnn - {gitops-title} 1.12.0 security update advisory
+
+Issued: 2024-03-14
```
```suggestion
Issued: 2024-03-20
```

Dhruv-Soni11 replied:
Fixed.

Srivaralakshmi commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,142 @@
+
+The list of security fixes that are included in this release is documented in the following advisory:
+
+* link:https://access.redhat.com/errata/RHSA-XXXX:NNNN[RHSA-xxxx:nnnn]
```
```suggestion
* link:https://access.redhat.com/errata/RHSA-2024:1441[RHSA-2024:1441]
```

Dhruv-Soni11 replied:
Fixed.

Srivaralakshmi commented on modules/gitops-release-notes-1-12-0.adoc:
```diff
@@ -0,0 +1,142 @@
+$ oc describe deployment gitops-operator-controller-manager -n openshift-gitops-operator
+----
+
+*The Title and errata number will change after the errata state moves from PUSH STATE to SHIPPED LIVE. The current entry is simply a placeholder at the moment*.
```
```suggestion

```
Remove this.

Dhruv-Soni11 replied:
Fixed.

Srivaralakshmi reviewed (commented):
@Dhruv-Soni11 Provided errata info for the embargoed advisory. PTAl and address it.

Srivaralakshmi commented:
LGTM

Srivaralakshmi commented:
/cherrypick enterprise-4.12

Srivaralakshmi commented:
/cherrypick gitops-docs-1.12
